microbench/edit_scalability.cpp

```diff
@@ -1,121 +1,193 @@
-  std::chrono::high_resolution_clock::time_point t1;
-
-  uint64_t num_nodes;
-  uint64_t num_edges;
-
-  t0             = std::chrono::high_resolution_clock::now();
-  auto edge_list = el_file_to_rand_vec_edge(inputFile, num_nodes, num_edges);
-  t1             = std::chrono::high_resolution_clock::now();
-  auto diff      = std::chrono::duration<uint64_t, std::nano>(t1 - t0).count();
-  std::cout << "File Read Time (ns):\t" << diff << std::endl;
-
-  t0 = std::chrono::high_resolution_clock::now();
-  switch (gtype) {
-  case LS_CIS:
-    run_p_evolve<LS_CSR_LOCK_OUT>(statsfn,
-                                  "LS_CIS " + std::to_string(numGThreads), c, 8,
-                                  num_nodes, num_edges, edge_list, lcis_evo);
-    break;
-  case LS_COO:
-    run_p_evolve<LS_CSR_LOCK_OUT>(statsfn,
-                                  "LS_COO " + std::to_string(numGThreads), c, 8,
-                                  num_nodes, num_edges, edge_list, lcoo_evo);
-    break;
-  case LS_CPR:
-    run_p_evolve<LS_CSR_LOCK_OUT>(statsfn,
-                                  "LS_CPR " + std::to_string(numGThreads), c, 8,
-                                  num_nodes, num_edges, edge_list, lcpo_evo);
-    break;
-  case LS_CSR:
-    run_p_evolve<LS_CSR_LOCK_OUT>(statsfn,
-                                  "LS_CSR " + std::to_string(numGThreads), c, 8,
-                                  num_nodes, num_edges, edge_list, lclo_evo);
+#include <fstream>
+#include <iostream>
+#include <mutex>
+#include <sstream>
+#include <utility>
+#include <vector>
+using namespace std;
+
+#include "algo.hpp"
+#include "counter.hpp"
+#include "galois/Galois.h"
+#include "graph.hpp"
+
+static const char* name = "Edit Scalability Benchmarking Suite";
+static const char* desc = "Creates graphs from files in order to see how"
+                          " edits impact algorithm performance.";
+
+/*
+ * Command-line args
+ */
+
+cl::opt<string> input_file_path(cl::Positional, cl::desc("[input file]"),
+                                cl::init("-"));
+
+cl::opt<size_t>
+    ingest_threads("ingest-threads",
+                   cl::desc("number of threads used for ingesting edges"),
+                   cl::init(1));
+
+cl::opt<size_t> algo_threads("algo-threads",
+                             cl::desc("number of threads used for algorithm"),
+                             cl::init(1));
+
+cl::opt<uint64_t> num_vertices("num-vertices",
+                               cl::desc("number of vertices in the graph"),
+                               cl::Required);
+
+enum GraphType { lscsr, morph };
+
+cl::opt<GraphType>
+    graph_type("graph,g", cl::desc("Choose graph representation:"),
+               cl::init(lscsr),
+               cl::values(clEnumValN(lscsr, "lscsr", "log-structured CSR"),
+                          clEnumValN(morph, "morph", "Galois MorphGraph")));
+
+/*
+ * Algo args
+ */
+
+enum AlgoName { nop, sssp_bfs };
+
+cl::opt<AlgoName> algo_name(
```

Comment:
Let's use a different library for our command line arguments, using llvm has caused a lot of problems. Use getopt.

Reply:
no pls